### DISCUSSION TITLE
How to perform the "nRF_DTM" on macOS?

### DISCUSSION THREAD

**Discussion by hsu772** [olleheugene/nRF-DTM#9] [Q&A]
Hi Sir,

I can download the "Release" file from Github web site for Window/Linux/macOS. 
I un-zip the file and find the file "nRF_DTM".
But I don't know how to perform it on macOS.

I tried to double click the "nRF_DTM" as the app file, I also change the access permissions from terminal with "chmod".
But they are all functionless.
Could you teach how to perform the "nRF_DTM" on macOS?

PS: Windows version works well on Win 10/11.

macOS: Monterey 12.7.4
CPU: Intel i7

Thank you,
Hsu

**Comment by olleheugene** [comment]
Hello Hsu,
Thanks for using the tool.
Usually, I use it on the terminal program directly.
But I don't think any problem even if run the tool on the Find.
I used a MacBook Pro (Intel-i5) with the latest OS (maybe Sonoma).
I will look into this and get back to you soon. 
For a while, could you test with [v2.6.1](https://github.com/olleheugene/nRF-DTM/releases/tag/v2.6.1)?

Thanks,
Eugene

**Reply by hsu772** [reply]
Hi Eugene,

Thanks for the quick response.
I downloaded the v2.6.1 and I occurred the same result "Permission denied" when I tried to perform at terminal.

When I tried to perform "nRF_DTM_x64_macOS" on Find app, it shows this app is from an unidentified developer and terminated the app.

Below result test at terminal and result also same as "nRF_DTM-master", sorry to not high light in first time.

Thank you,
Han

---- test start -----
Hande-MacBook-Pro:macOS han$ ./nRF_DTM_x64_macOS
**-bash: ./nRF_DTM_x64_macOS: Permission denied**
Hande-MacBook-Pro:macOS han$ sudo su
Password:
sh-3.2# ls -al
total 68216
drwxr-xr-x@ 3 han  staff        96  2  6  2024 .
drwxr-xr-x@ 7 han  staff       224  2  6  2024 ..
-rw-r--r--@ 1 han  staff  34922877  2  6  2024 nRF_DTM_x64_macOS
sh-3.2# ./nRF_DTM_x64_macOS
s**h: ./nRF_DTM_x64_macOS: Permission denied**
**sh-3.2# chmod u+x nRF_DTM_x64_macOS** 
sh-3.2# ls -al
total 68216
drwxr-xr-x@ 3 han  staff        96  2  6  2024 .
drwxr-xr-x@ 7 han  staff       224  2  6  2024 ..
-rwxr--r--@ 1 han  staff  34922877  2  6  2024 nRF_DTM_x64_macOS
sh-3.2# ./nRF_DTM_x64_macOS
Killed: 9

---- test end -----

**Reply by olleheugene** [reply]
Hi Han,

Sorry for the inconvenience.
I didn't know it had trouble to run from other users.
This is not officially registered in the Apple developer program.
So, to perform the tool you need to follow the way that is described in the link.
Can you follow the below link?
[Open a Mac app from an unidentified developer](https://support.apple.com/en-ca/guide/mac-help/-mh40616/mac)

Thanks,
Eugene
